src/monodroid/jni/logger.c

```diff
@@ -171,6 +173,10 @@ init_logging_categories ()
 			log_categories  |= LOG_LREF;
 			light_lref       = 1;
 		}
+
+		if (!strncmp (arg, "timing=bare", 11)) {
```

Comment:
Shouldn't this be `else if` and part of the above tree?

Reply:
I left it away from it for better readability. The whole loop is not efficient anyway and runs only when the log property is set.

Maybe we should handle the lref, gref and timing cases separately without using `CATEGORY` define for them.

Reply:
I'm not sure what that would look like, short of introducing new separate system properties for lref & gref logging, e.g. `debug.mono.log.gref`.

This introduces a different set of complications around how to properly support system properties going forward...

Reply:
I mean just not using the define for them, but keep the `debug.mono.log` meaning as before. (if we care about speed here)

So replacing for example this line:

    CATEGORY ("gref", LOG_GREF);

with something like:

    if (!strncmp (arg, "gref", 4)) {
        log_categories |= LOG_GREF;
        switch (arg[4]) {
            case '=':
                gref_file = arg + 5;
                break;
            case '-':
                light_gref = 1;
                break;
        }
    }